pkg/controllers/namespace/subnet_handler.go

```diff
@@ -0,0 +1,336 @@
+			Name:      subnetName,
+			Namespace: ns,
+			Annotations: map[string]string{
+				servicecommon.AnnotationAssociatedResource: associatedName,
+			},
+		},
+		Spec: v1alpha1.SubnetSpec{
+			VPCName:             vpcFullName,
+			EnableVLANExtension: true,
+		},
+	}
+
+	// Initialize subnetCR from nsxSubnet if available
+	if nsxSubnet != nil {
+		r.mapNSXSubnetToSubnetCR(subnetCR, nsxSubnet)
+	} else {
+		// Use default values if nsxSubnet is not available
+		subnetCR.Spec.AccessMode = v1alpha1.AccessMode(v1alpha1.AccessModePublic)
+		subnetCR.Spec.IPv4SubnetSize = 64
+		subnetCR.Spec.SubnetDHCPConfig.Mode = v1alpha1.DHCPConfigMode(v1alpha1.DHCPConfigModeDeactivated)
+	}
+
+	return subnetCR
+}
+
+// mapNSXSubnetToSubnetCR maps NSX subnet properties to Subnet CR properties
+func (r *NamespaceReconciler) mapNSXSubnetToSubnetCR(subnetCR *v1alpha1.Subnet, nsxSubnet *model.VpcSubnet) {
+	// Map AccessMode
+	if nsxSubnet.AccessMode != nil {
+		accessMode := *nsxSubnet.AccessMode
+		// Convert from NSX format to v1alpha1 format
+		if accessMode == "Private_TGW" {
+			subnetCR.Spec.AccessMode = v1alpha1.AccessMode(v1alpha1.AccessModeProject)
+		} else {
+			subnetCR.Spec.AccessMode = v1alpha1.AccessMode(accessMode)
+		}
+	} else {
+		subnetCR.Spec.AccessMode = v1alpha1.AccessMode(v1alpha1.AccessModePublic)
+	}
+
+	// Map IPv4SubnetSize
+	if nsxSubnet.Ipv4SubnetSize != nil {
+		subnetCR.Spec.IPv4SubnetSize = int(*nsxSubnet.Ipv4SubnetSize)
+	} else {
+		subnetCR.Spec.IPv4SubnetSize = 64
+	}
+
+	// Map IPAddresses
+	subnetCR.Spec.IPAddresses = nsxSubnet.IpAddresses
+
+	// Map SubnetDHCPConfig
+	if nsxSubnet.SubnetDhcpConfig != nil && nsxSubnet.SubnetDhcpConfig.Mode != nil {
+		dhcpMode := *nsxSubnet.SubnetDhcpConfig.Mode
+		// Convert from NSX format to v1alpha1 format
+		switch dhcpMode {
+		case "DHCP_SERVER":
+			subnetCR.Spec.SubnetDHCPConfig.Mode = v1alpha1.DHCPConfigMode(v1alpha1.DHCPConfigModeServer)
+		case "DHCP_RELAY":
+			subnetCR.Spec.SubnetDHCPConfig.Mode = v1alpha1.DHCPConfigMode(v1alpha1.DHCPConfigModeRelay)
+		default:
+			subnetCR.Spec.SubnetDHCPConfig.Mode = v1alpha1.DHCPConfigMode(v1alpha1.DHCPConfigModeDeactivated)
+		}
+	} else {
+		subnetCR.Spec.SubnetDHCPConfig.Mode = v1alpha1.DHCPConfigMode(v1alpha1.DHCPConfigModeDeactivated)
+	}
+}
+
+// createSubnetCRInK8s creates the Subnet CR in Kubernetes
+func (r *NamespaceReconciler) createSubnetCRInK8s(ctx context.Context, subnetCR *v1alpha1.Subnet, subnetName string) error {
+	err := r.Client.Create(ctx, subnetCR)
+	if err != nil {
+		// If the Subnet CR already exists with the same name, try using generateName
+		if strings.Contains(err.Error(), "already exists") {
+			log.Info("Subnet CR with the same name already exists, using generateName",
+				"Namespace", subnetCR.Namespace, "Name", subnetName)
+
+			// Create a new Subnet CR with generateName
+			// subnetCR.ObjectMeta.Name will be subnetName + "-" + randomSuffix
+			subnetCR.ObjectMeta.GenerateName = subnetName + "-"
```

Comment:
Have you try that if the Subnet CR has both "name" and "generateName", can it be created? If not, here we need to remove "name" from the Subnet CR.

Reply:
yes, should remove name, fixed